### ISSUE TITLE
[MacOS Mojave 10.14.6] Issue with sf::Event::KeyPressed

### ISSUE THREAD

**Issue by ImaGrimGrimGuy** [SFML/SFML#1618]
Thanks for raising your issue here! 🙂

Before you submit the issue however, we'd like you to consider the follow points.

* We like to use the issue tracker for confirmed issues.
* If you're stuck with SFML, please use [the forum](https://en.sfml-dev.org/forums/index.php#c3) to get help.

----

## Subject of the issue

Describe your issue here.

## Your environment

* Your OS / distro / window manager used
* Your version of SFML (2.5.0, git master, etc)
* Your compiler and compiler version used
* Special compiler flags used

## Steps to reproduce

Tell us how to reproduce this issue. Please provide a [minimal, complete and verifiable example](https://stackoverflow.com/help/mcve), you can use the follow template as a start:

```cpp
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode(1280, 720), "Minimal, complete and verifiable example");
    window.setFramerateLimit(60);

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }
}
```

## Expected behavior

Tell us what should happen

## Actual behavior

Tell us what happens instead

**Comment by ImaGrimGrimGuy** [comment]
Hi, I've run into an issue with keyboard input from a MacBook Pro.  It seems that more than one person has had issues with their keyboards.

I've got an event polling loop which detects various events with a switch statement, and It seems that not only does the RETURN key not produce any event, it also prevents other keyboard input from being detected until the window is clicked on again. Of note is the fact that the ENTER key does not produce this error, and the error itself is specific to keyboard input -eg clicking, scrolling, and moving the mouse are all still detected without issue.

A few facts about how this is being built:
-I'm using Xcode
-I'm using a MacBook Pro with MacOS Mojave 10.14.6 installed
-I'm building the project using XCode's APP-Creator settings.

Here is a snippet of code where I've managed to Isolate the issue:

void ZivixGraphics::update() {
        sf::Vector2i mpos = sf::Mouse::getPosition(*m_Window);
        mouseMove(mpos.x, mpos.y);

        sf::Event event;
        while (getWindow()->pollEvent(event)) {
            switch (event.type) {
                case sf::Event::KeyPressed:
                    std::cout << "Detected a " << std::to_string(event.key.code) << " Key Input!";
                    break;
                default:
                    std::cout << "Detected a " << std::to_string(event.type) << " Event!";
                    break;   
            }
        }
}
